firebase-common/src/main/java/com/google/firebase/heartbeatinfo/HeartBeatInfoStorage.java

```diff
@@ -93,9 +93,39 @@ synchronized void deleteAllHeartBeats() {
                 entry.getKey(), new ArrayList<String>((Set<String>) entry.getValue())));
       }
     }
+    updateGlobalHeartBeat(System.currentTimeMillis());
     return heartBeatResults;
   }
 
+  synchronized void postHeartBeatCleanUp() {
+    String dateString = getFormattedDate(System.currentTimeMillis());
+    firebaseSharedPreferences.edit().putString(LAST_STORED_DATE, dateString).apply();
+
+    // Find stored heartbeat and clear it.
+    String userAgentString = "";
+    for (Map.Entry<String, ?> entry : firebaseSharedPreferences.getAll().entrySet()) {
+      if (entry.getValue() instanceof Set) {
+        Set<String> dateSet = (Set<String>) entry.getValue();
+        for (String date : dateSet) {
+          if (dateString.equals(date)) {
+            userAgentString = entry.getKey();
```

Comment:
you're not breaking out of the outer loop, which is what I think we want. To avoid this problem entirely, consider extracting all of this logic into a method and use `return` instead of a `break`.

Reply:
hmm in the beginning there is an if condition on the userAgentString if its not null.. I break.. So it would break out of all the loops

Reply:
I see, but still for readability it would be good to extract this logic into a separate method and give it a meaningful name.